dace/codegen/targets/xilinx.py

```diff
@@ -134,298 +68,201 @@ def cmake_options():
 
     @staticmethod
-    def detect_memory_widths(subgraphs):
-        stack = []
-        for sg in subgraphs:
-            stack += [(n, sg) for n in sg.nodes()]
-        memory_widths = {}
-        seen = set()
-        while len(stack) > 0:
-            node, graph = stack.pop()
-            if isinstance(node, dace.graph.nodes.NestedSDFG):
-                for state in node.sdfg.states():
-                    stack += [(n, state) for n in state.nodes()]
-            elif isinstance(node, dace.graph.nodes.AccessNode):
-                if node in seen:
-                    continue
-                seen.add(node)
-                nodedesc = node.desc(graph)
-                for edge in graph.all_edges(node):
-                    if (isinstance(edge.data, dace.memlet.EmptyMemlet)
-                            or edge.data.data is None):
-                        continue
-                    if node.data not in memory_widths:
-                        if (isinstance(nodedesc, dace.data.Stream)
-                                and nodedesc.veclen != edge.data.veclen):
-                            raise ValueError(
-                                "Vector length on memlet {} ({}) doesn't "
-                                "match vector length of {} ({})".format(
-                                    edge.data, edge.data.veclen, node.data,
-                                    nodedesc.veclen))
-                        memory_widths[node.data] = edge.data.veclen
-                    else:
-                        if memory_widths[node.data] != edge.data.veclen:
-                            raise dace.codegen.codegen.CodegenError(
-                                "Inconsistent vector length "
-                                "on FPGA for \"{}\": got {}, had {}".format(
-                                    node.data, edge.data.veclen,
-                                    memory_widths[node.data]))
-        return memory_widths
+    def generate_pipeline_loop_post(kernel_stream, sdfg, state_id, node):
+        kernel_stream.write("#pragma HLS PIPELINE II=1", sdfg, state_id, node)
 
-    def generate_kernel(self, sdfg, state, kernel_name, subgraphs,
-                        function_stream, callsite_stream):
+    @staticmethod
+    def generate_flatten_loop_pre(kernel_stream, sdfg, state_id, node):
+        pass
 
-        state_id = sdfg.node_id(state)
+    @staticmethod
+    def generate_flatten_loop_post(kernel_stream, sdfg, state_id, node):
+        kernel_stream.write("#pragma HLS LOOP_FLATTEN")
 
-        (global_data_params, top_level_local_data, subgraph_params,
-         scalar_parameters, symbol_parameters,
-         nested_global_transients) = type(self).make_parameters(
-             sdfg, state, subgraphs)
+    @staticmethod
+    def make_read(defined_type, type_str, var_name, vector_length, expr,
+                  index):
+        if defined_type in [DefinedType.Stream, DefinedType.StreamView]:
+            return "{}.pop()".format(expr)
```

Comment:
Should be uppercase "Pop"

Reply:
Everything (pop/push) seems to be defined with lowercase

Reply:
Yes I can fix this

Reply:
Implemented with eccb769